.github/workflows/go-integration.yml

```diff
@@ -0,0 +1,91 @@
+name: Go End-to-End
+
+on:
+  push:
+    branches-ignore:
+      - master
+  pull_request:
+    branches-ignore:
+      - gh-pages
+
+env:
+  GOPATH: /actions-runner/go
+  GOCACHE: /actions-runner/gocache
+  GITHUB_WORKSPACE: /actions-runner/go/src/github.com/G-Research/armada
+  HOME: /actions-runner
+
+jobs:
+  start-runner:
+    name: Start runner in EC2
+    if: github.repository == 'G-Research/armada'
+    runs-on: ubuntu-latest
+    outputs:
+      runs-on: ${{ steps.start-ec2-runner.outputs.label }}
+      ec2-instance-id: ${{ steps.start-ec2-runner.outputs.ec2-instance-id }}
+    steps:
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: ${{ secrets.AWS_REGION }}
+      - name: Start EC2 runner
+        id: start-ec2-runner
+        uses: machulav/ec2-github-runner@v2
+        with:
+          mode: start
+          github-token: ${{ secrets.GH_PERSONAL_ACCESS_TOKEN }}
+          ec2-image-id: ami-0d3d3ae42ef82e511
+          ec2-instance-type: c5.2xlarge
+          subnet-id: subnet-06ae121731a8cddc2
+          security-group-id: sg-0a0571fdd1322701f
+          aws-resource-tags: >
+            [
+              {"Key": "Name", "Value": "ec2-github-runner"},
+              {"Key": "GitHubRepository", "Value": "${{ github.repository }}"}
+            ]
+  integration-tests:
+      needs: start-runner
+      if: github.repository == 'G-Research/armada'
+      runs-on: ${{ needs.start-runner.outputs.runs-on }}
+      strategy:
+        fail-fast: false
+        matrix:
+          go: [ '1.18' ]
+      steps:
+        - uses: actions/checkout@v3
+        - uses: ./.github/workflows/go-setup
+        - run: make build-ci
+        - run: make tests-e2e-setup
+        - run: make tests-e2e-no-setup
+          env:
+            INTEGRATION_ENABLED: true
+        - run: make junit-report
+        - name: Upload junit report
+          uses: actions/upload-artifact@v2.2.4
+          with:
+            name: junit.xml
+            path: test_reports/junit.xml
+  stop-runner:
+      name: Stop self-hosted runner
+      if: always()
```

Comment:
Why would you always stop?

Reply:
See if the comments I just added makes it clearer :)